schema_based_on_diagram.sql

```diff
@@ -0,0 +1,52 @@
+CREATE TABLE patients (
+  id INT,
+  name VARCHAR,
+  date_of_birth DATE,
+  PRIMARY KEY (id)
+);
+
+CREATE TABLE medical_histories (
+  id INT,
+  admitted_at TIMESTAMP,
+  patient_id INT,
+  status VARCHAR,
+  PRIMARY KEY (id),
+  FOREIGN KEY (patient_id) REFERENCES patients(id)
+);
+
+CREATE TABLE treatments (
+  id INT,
+  type VARCHAR,
+  name VARCHAR,
+  PRIMARY KEY (id)
+);
+
+CREATE TABLE history_treatment (
+  id SERIAL PRIMARY KEY,
+  medical_histories_id INT,
+  treatment_id INT,
+  FOREIGN KEY (medical_histories_id) REFERENCES medical_histories(id),
+  FOREIGN KEY (treatment_id) REFERENCES treatments(id),
+);
+
+CREATE TABLE invoices (
+  id INT,
+  total_amount DECIMAL,
+  generated_at TIMESTAMP,
+  payed_at TIMESTAMP,
+  medical_history__id INT,
+  PRIMARY KEY (id),
+  FOREIGN KEY (medical_history__id) REFERENCES medical_histories(id)
+);
+
+CREATE TABLE invoice_items (
+  id INT,
+  unit_price DECIMAL,
+  quantity INT,
+  total_price DECIMAL,
+  invoice_id INT,
+  treatment_id INT,
+  PRIMARY KEY (id),
+  FOREIGN KEY (invoice_id) REFERENCES invoices(id),
+  FOREIGN KEY (treatment_id) REFERENCES treatments(id)
+);
```

Comment:
- [ ] Based on the project requirements, kindly create indexes on all the six `foreign keys` that you have. Creating indexes on foreign keys is useful for finding all source rows that reference a target row. If the referenced column(s) are changed frequently, it might be wise to add an index to the referencing column(s) so that referential actions associated with the foreign key constraint can be performed more efficiently. Kindly read this [doc](https://www.postgresql.org/docs/current/sql-createtable.html) for further explanation.